### PR TITLE
Refactor: Improve filesystem connector property tests

### PR DESCRIPTION
**Pull request by ahrav**
Add comprehensive property tests for the `FilesystemConnector`. This includes:
- Strategies for generating adversarial file paths with mixed components, Unicode, and edge-case prefixes.
- Strategies for creating fixtures with explicit empty directories and a variety of file structures.
- A utility function `collect_keys_from_cursor` to robustly drain pages from a cursor, with a step guard against infinite loops.
- Test cases that exercise resumability with varying page sizes and simulated process restarts.

### CODE REVIEW

File: crates/gossip-connectors/src/filesystem_tests.rs [hunk 1]
```diff
@@ -2232,6 +2346,43 @@ mod prop {
...
+    ) -> Vec<Vec<u8>> {
+        let mut conn = FilesystemConnector::new(root).with_tokens(with_tokens);
+        let budgets = small_page_budgets(page_size.max(1));
+        let mut keys = Vec::new();
+        let mut steps = 0usize;
+        loop {
+            assert!(
+                steps < 4096,
+                "pagination did not terminate while collecting from cursor"
+            );
```
[the remark this answers is not in the document]
Comment (ahrav) [reply crates/gossip-connectors/src/filesystem_tests.rs]: Good catch. Refactored `collect_keys_from_cursor` to return `Result<Vec<Vec<u8>>, TestCaseError>` and replaced the `assert!` with `prop_assert!`. The call site in `corrupt_token_falls_back_to_last_key` now propagates with `?`.

This ensures proptest classifies guard violations as failures (with shrinking) rather than errors, and avoids process termination under `panic = "abort"` profiles.

File: crates/gossip-connectors/src/filesystem_tests.rs [hunk 2]
```diff
@@ -2454,6 +2605,278 @@ mod prop {
...
+        ) {
+            // Range safety: emitted keys must stay inside [start, end).
+            let (files, empty_dirs) = fixture;
+            let (start_bound, end_bound) = if raw_start <= raw_end {
+                (raw_start, raw_end)
+            } else {
+                (raw_end, raw_start)
+            };
+            let shard = ShardSpec::try_with_range(start_bound.as_slice(), end_bound.as_slice())
+                .expect("valid shard bounds");
```
[the remark this answers is not in the document]
Comment (ahrav) [reply crates/gossip-connectors/src/filesystem_tests.rs]: Investigated the empty/empty case specifically: `ShardSpec::try_with_range(b"", b"")` is actually valid — the validation at `shard_spec.rs:452` requires *both* bounds to be non-empty before checking `start >= end`, so empty/empty passes validation and returns `Ok(...)`. There is also an existing test confirming this (`ShardSpec::try_with_range(b"", b"").unwrap()` at line 1944).

That said, the broader concern about the `.expect()` was valid for **equal non-empty bounds** (covered by the codex-connector comment above), and the fix — replacing `.expect()` with a graceful `let Ok(shard) = ... else { return Ok(()); }` — handles all invalid-bounds cases, including any future invariant tightening.
